src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -226,12 +219,28 @@ public FeatureInfo(Feature feature, FeatureDiagnosticData diagnostics)
 
             public FeatureDiagnosticData Diagnostics { get; }
             public Feature Feature { get; }
-            public IReadOnlyList<FeatureStartupTaskController> TaskControllers { get; set; }
+            public IReadOnlyList<FeatureStartupTaskController> TaskControllers => taskControllers;
+
+            public void InitializeFrom(FeatureConfigurationContext featureConfigurationContext)
+            {
+                Feature.SetupFeature(featureConfigurationContext);
+                var featureStartupTasks = new List<string>();
+                foreach (var controller in featureConfigurationContext.TaskControllers)
+                {
+                    taskControllers.Add(controller);
+                    featureStartupTasks.Add(controller.Name);
+                }
+                featureConfigurationContext.TaskControllers.Clear();
```

Comment:
I like this in general but I'd say .Clear should be done outside of this method?